client/web/src/search/results/SearchResultsInfoBar.tsx

```diff
@@ -64,55 +70,79 @@ const QuotesInterpretedLiterallyNotice: React.FunctionComponent<SearchResultsInf
  * The info bar shown over the search results list that displays metadata
  * and a few actions like expand all and save query
  */
-export const SearchResultsInfoBar: React.FunctionComponent<SearchResultsInfoBarProps> = props => (
-    <div className={classNames(props.className, 'search-results-info-bar')} data-testid="results-info-bar">
-        <small className="search-results-info-bar__row">
-            {props.stats}
-            <QuotesInterpretedLiterallyNotice {...props} />
+export const SearchResultsInfoBar: React.FunctionComponent<SearchResultsInfoBarProps> = props => {
+    const CreateCodeMonitorButton = useMemo(() => {
+        if (!props.enableCodeMonitoring || !props.query) {
+            return null
+        }
+        const globalTypeFilterInQuery = findFilter(props.query, 'type', FilterKind.Global)
+        const globalTypeFilterValue =
+            globalTypeFilterInQuery?.value?.type === 'literal'
+                ? globalTypeFilterInQuery.value.value
+                : globalTypeFilterInQuery?.value?.type === 'quoted'
+                ? globalTypeFilterInQuery.value.quotedValue
+                : undefined
+        const canCreateMonitorFromQuery = globalTypeFilterValue === 'diff' || globalTypeFilterValue === 'commit'
+        if (!canCreateMonitorFromQuery) {
+            return null
+        }
+        const searchParameters = new URLSearchParams(props.location.search)
+        searchParameters.set('trigger-query', `${props.query} patterntype:${props.patternType}`)
+        const toURL = canCreateMonitorFromQuery ? `/code-monitoring/new?${searchParameters.toString()}` : ''
```

Comment:
You could simplify this line to just `/code-monitoring/new?${searchParameters.toString()}`, because `canCreateMonitorFromQuery` is always `true` here due to the `if (!canCreateMonitorFromQuery) { return null }` guard above.

Reply:
Ah yup, good catch, thanks! This is a leftover from an intermediary state where that condition wasn't always `true`

```suggestion
        const toURL = `/code-monitoring/new?${searchParameters.toString()}`
```